File: ERP/HomeFloory/Entities/Korisnik.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace HomeFloory.Models;

public partial class Korisnik
{
    public decimal IdKorisnik { get; set; }

    public string? Ime { get; set; }

    public string? Prezime { get; set; }

    public DateTime? DatumRodjenja { get; set; }

    public string? Kontakt { get; set; }

    public string? Email { get; set; }

    public string? Lozinka { get; set; }

    public decimal? IdAdresaIsporuke { get; set; }

    [DefaultValue(1)]
    public decimal IdUloga { get; set; }

    public virtual AdresaIsporuke? IdAdresaIsporukeNavigation { get; set; }

    public virtual Uloga IdUlogaNavigation { get; set; } = null!;

    public virtual ICollection<Placanje> Placanja { get; } = new List<Placanje>();
}

File: Frontend/src/app/shared/models/korisnik.ts
export interface Korisnik {
    idKorisnik: number
    ime: string
    prezime: string
    datumRodjenja: string
    kontakt: string
    email: string
    lozinka: string
    idAdresaIsporuke: number
    idUloga: 1
    token: string
}

export interface Kor {
    email: string
    lozinka: string
    token: string
}

export interface AdresaIsporuke {
    idAdresaIsporuke?: number
    grad: string | null
    drzava: string | null
    ulica: string | null
    postanskiBroj: string | null
}

export interface IdUlogaNavigation {
    idUloga: number
    uloga1: string
    korisnici: Korisnici | undefined[]
}

export interface Korisnici {
    idKorisnik: number
    ime: string
    prezime: string
    datumRodjenja: string
    kontakt: string
    email: string
    lozinka: string
    idAdresaIsporuke: any
    idUloga: number
    idAdresaIsporukeNavigation: any
    idUlogaNavigation: any
    placanja: any[]
}